src/WebJobs.Script/Diagnostics/DiagnosticEventLoggerExtensions.cs

```diff
@@ -20,7 +20,7 @@ public static void LogDiagnosticEvent(this ILogger logger, LogLevel level, int e
                 { ScriptConstants.ErrorCodeKey, errorCode }
             };
 
-            logger.Log(level, eventId, stateDict, exception, (state, ex) => message);
+            logger.Log(level, new EventId(0, $"DiagnosticEvent-{errorCode}"), stateDict, exception, (state, ex) => message);
```

Comment:
Should the event ID be passed in?

```suggestion
            logger.Log(level, new EventId(eventId, $"DiagnosticEvent-{errorCode}"), stateDict, exception, (state, ex) => message);
```

Reply:
event id is hard coded to 0, Not really used anywhere. The documentation for diagnostic event also suggests hardcoding it to 0 as we mainly use error code as the key identifier. Would want to keep it as is to avoid confusion.

Reply:
I am not sure this is a good use of `EventId`.

1. This looks like a log breaking change. We are previously logging with a non-zero event ID, and now it will be 0.
2. These are unique diagnostic events after all, they should have their own event ID. Why do they need to be the same?

Reply:
Looking at usage, I don't see anyone using a non-zero eventId. In that case, I recommend we remove `eventId` from the API entirely.